src/main/kotlin/io/github/gradlenexus/publishplugin/DefaultNexusRepositoryContainer.kt

```diff
@@ -20,16 +20,14 @@ import groovy.lang.Closure
 import org.gradle.api.Action
 import org.gradle.api.NamedDomainObjectContainer
 import org.gradle.api.internal.NamedDomainObjectContainerConfigureDelegate
-import org.gradle.util.ConfigureUtil
 import java.net.URI
 import javax.inject.Inject
 
-@Suppress("UnstableApiUsage")
 internal open class DefaultNexusRepositoryContainer @Inject constructor(
     delegate: NamedDomainObjectContainer<NexusRepository>
 ) : NexusRepositoryContainer, NamedDomainObjectContainer<NexusRepository> by delegate {
 
-    override fun sonatype(): NexusRepository = sonatype {}
+    override fun sonatype(): NexusRepository = sonatype(Action {})
```

Comment:
Is this change needed?

Reply:
It would save a WTF moment if you added that into the code comment 😉

Reply:
I think it's ok if you're not looking at the diff. And if you try to "fix" it, you'll get a compile error. What would be the comment?

Reply:
For instance:

```
// sonatype {} here does not work with Kotlin 1.3 for unknown reason
```

Reply:
https://github.com/gradle-nexus/publish-plugin/pull/168/commits/20f22c49c56f994ba839eede2befd54bc75d84cd

Reply:
Also I think it's exactly because of this line:
https://github.com/gradle/gradle/blob/v8.0.0/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslCompilerPlugins.kt

which was different in

https://github.com/gradle/gradle/blob/v6.9.0/subprojects/kotlin-dsl-plugins/src/main/kotlin/org/gradle/kotlin/dsl/plugins/dsl/KotlinDslCompilerPlugins.kt#L61

Reply:
I tried enabling new inference with: `kotlinOptions.freeCompilerArgs += "-XXLanguage:+NewInference"`, but I still get an error and a huge warning:
```
> Task :compileKotlin FAILED
w: ATTENTION!
This build uses unsafe internal compiler arguments:

-XXLanguage:+NewInference

This mode is not recommended for production use,
as no stability/compatibility guarantees are given on
compiler or generated code. Use it at your own risk!

e: file:///P:/projects/contrib/github-nexus-publish-plugin/src/main/kotlin/io/github/gradlenexus/publishplugin/DefaultNexusRepositoryContainer.kt:30:57 Type mismatch: inferred type is () -> Unit but Action<in NexusRepository> was expected

FAILURE: Build failed with an exception.
```
So lets stick with these two tiny syntax changes with comments.